### ISSUE TITLE
MASKL(0)/MASKR(0) return -1 instead of 0

### ISSUE THREAD

**Issue by jeanPerier** [llvm/llvm-project#56706]
Repro:
```f90
  integer :: n
  n = 0
  ! Expects 0, gets -1
  print *, maskl(n)
  print *, maskr(n)
end
```

MASK intrinsic lowering was just implemented in https://github.com/llvm/llvm-project/commit/71ee3570f46cbbbaf08f98d4338ff0946e7dd2b8. The results in the zero case appears to be wrong.

**Comment by tarunprabhu** [comment]
I was able to reproduce this. I will look into it now.

**Comment by xgupta** [comment]
Checked today's trunk, and it is fixed. Shall we close the issue?

**Comment by tarunprabhu** [comment]
This issue was fixed in [c1f65df](https://github.com/llvm/llvm-project/commit/c1f65df19c0f7324db8a7ec235308aee55fe22ef)
